src/backend/commands/copy.c

```diff
@@ -7955,9 +7955,21 @@ GetTargetSeg(GpDistributionData *distData, Datum *baseValues, bool *baseNulls)
 	return target_seg;
 }
 
+static void
+close_program_pipes_on_reset(void *arg)
+{
+	if (!IsAbortInProgress())
+		return;
+
+	CopyState	cstate = arg;
+
+	close_program_pipes(cstate, false);
```

Comment:
Maybe reuse web-table approach from `destroy_execute_handle` to avoid patching pclose_with_stderr?

```suggestion
	close_program_pipes(cstate, false);
	if (cstate->program_pipes->pipes[EXEC_DATA_P] != -1)
		close(cstate->program_pipes->pipes[EXEC_DATA_P]);

	/* We don't bother reading possible error message from the pipe */
	if (cstate->program_pipes->pipes[EXEC_ERR_P] != -1)
		close(cstate->program_pipes->pipes[EXEC_ERR_P]);
	if (h->pid != -1)
	{
#ifndef WIN32
		int			status;
		waitpid(cstate->program_pipes->pid, &status, 0);
#endif
	}
```

Reply:
Closed as over-optimization.